### ISSUE TITLE
Plugin fails with milestone and RC scala versions

### ISSUE THREAD

**Issue by fiadliel** [ThoughtWorksInc/sbt-api-mappings#5]
When using this plugin, and cross-compiling against scala 2.12.0-RC2, I get this error:

```
scala.MatchError: scala-compiler-2.12.0-RC2.jar (of class java.lang.String)
	at com.thoughtworks.sbtApiMappings.ApiMappings$$anonfun$projectSettings$3$$anonfun$apply$2.apply(ApiMappings.scala:41)
	at com.thoughtworks.sbtApiMappings.ApiMappings$$anonfun$projectSettings$3$$anonfun$apply$2.apply(ApiMappings.scala:39)
```

The regex `ScalaLibraryRegex` doesn't match M* or RC* versions.

**Comment by Atry** [comment]
What is the version of sbt-api-mappings that you are using?

**Comment by fiadliel** [comment]
It's version 0.2.4. The required code to reproduce is in https://gist.github.com/fiadliel/fc7d5138f659284d660a28037e144acc - but it's basically just set the scala version, add the plugin, and run `sbt doc`.

**Comment by Atry** [comment]
Thank you for your report. I will fix it soon. You can also create Pull Requests 😄

**Comment by 2m** [comment]
Just encountered this problem as well. And I was confused when I was not able to find `ScalaLibraryRegex` in the source code. Only now I have realized that you have just pushed the changes that revamped this plugin. :) Looking forward to the release.

**Comment by Atry** [comment]
Fixed in sbt-api-mappings 1.0.0
